aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,106 @@
+For features that require substantial setup, there are `Add[Service]` extension methods on [IServiceCollection](/dotnet/api/Microsoft.Extensions.DependencyInjection.IServiceCollection). A typical web app registers services for Entity Framework, Identity, and MVC:
 
 [!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=4,7,11&start=40&end=55)]
 
-Adding services to the services container makes them available within your application via [dependency injection](xref:fundamentals/dependency-injection).
-
-## Services Available in Startup
-
-ASP.NET Core dependency injection provides services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or its `Configure` method. The `ConfigureServices` method only takes an `IServiceCollection` parameter (but any registered service can be retrieved from this collection, so additional parameters are not necessary).
+## Services available in Startup
 
-Below are some of the services typically requested by `Startup` methods:
+ASP.NET Core [dependency injection](xref:fundamentals/dependency-injection) provides services during an app's startup. These services are requested by including the appropriate interface as a parameter on the `Startup` class's constructor or its `Configure` method. The `ConfigureServices` method only takes an `IServiceCollection` parameter. Any registered service can be retrieved from this collection, so additional parameters aren't necessary.
 
-* In the constructor:  `IHostingEnvironment`, `ILogger<Startup>`
-* In `ConfigureServices`:  `IServiceCollection`
-* In `Configure`:  `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`
+Some of the services typically requested by `Startup` methods:
 
-Any services added by the ``WebHostBuilder`` ``ConfigureServices`` method may be requested by the ``Startup`` class constructor or its ``Configure`` method. Use `WebHostBuilder` to provide any services you need during `Startup` methods.
+* In the constructor: `IHostingEnvironment`, `ILogger<Startup>`
+* In `ConfigureServices`: `IServiceCollection`
+* In `Configure`: `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`
 
 ## The Configure method
 
-The `Configure` method is used to specify how the ASP.NET application will respond to HTTP requests. The request pipeline is configured by adding [middleware](middleware.md) components to an `IApplicationBuilder` instance that is provided by dependency injection.
+The [Configure](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configure) method is used to specify how the app responds to HTTP requests. The request pipeline is configured by adding [middleware](xref:fundamentals/middleware) components to an [IApplicationBuilder](/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder) instance that's provided by [dependency injection](xref:fundamentals/dependency-injection).
```

Comment:
Subtle point: IApplicationBuilder is available to the Configure method but it's never registered in any container as a service. Hosting creates it and passes it to Configure directly.
https://github.com/aspnet/Hosting/blob/9f1e6607dd1b3d15bc6c42146629677c6b455fd0/src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs#L206-L219

Reply:
I think TMI on this one. @Rick-Anderson Thoughts?

Reply:
@guardrex  I think it's worth adding

Note: ` IApplicationBuilder` is available to the `Configure` method but it's never registered in any container as a service. Hosting creates it and passes it to `Configure` directly. See [WebHost BuildApplication](
https://github.com/aspnet/Hosting/blob/9f1e6607dd1b3d15bc6c42146629677c6b455fd0/src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs#L206-L219)